File: kotlin/maven/src/test/kotlin/example/utils/TargetEnvResolver.kt
package example.utils

// Data class to store environment-specific information
data class EnvInfo(
    val pageUrl: String,
    val baseUrl: String,
    val usersFeederFile: String,
    val productsFeederFile: String
)

object TargetEnvResolver {

    // Resolve environment-specific configuration based on the target environment
    fun resolveEnvironmentInfo(targetEnv: String): EnvInfo {
        return when (targetEnv) {
            "DEV" -> EnvInfo(
                pageUrl = "https://ecomm.gatling.io",
                baseUrl = "https://api-ecomm.gatling.io",
                usersFeederFile = "data/users_dev.json",
                productsFeederFile = "data/products_dev.csv"
            )
            else -> EnvInfo(
                pageUrl = "https://ecomm.gatling.io",
                baseUrl = "https://api-ecomm.gatling.io",
                usersFeederFile = "data/users_dev.json",
                productsFeederFile = "data/products_dev.csv"
            )
        }
    }
}

File: scala/maven/src/test/scala/example/utils/TargetEnvResolver.scala
package example.utils

object TargetEnvResolver {

  // Record to store environment-specific information
  case class EnvInfo(pageUrl: String, baseUrl: String, usersFeederFile: String, productsFeederFile: String)

  // Resolve environment-specific configuration based on the target environment
  def resolveEnvironmentInfo(targetEnv: String): EnvInfo = targetEnv match {
    case "DEV" =>
      EnvInfo("https://ecomm.gatling.io", "https://api-ecomm.gatling.io", "data/users_dev.json", "data/products_dev.csv")
    case _ =>
      EnvInfo("https://ecomm.gatling.io", "https://api-ecomm.gatling.io", "data/users_dev.json", "data/products_dev.csv")
  }
}
